core/src/main/scala/pipeflow/system/iso8601/IntervalDuration.scala

```diff
@@ -0,0 +1,77 @@
+package pipeflow.system.iso8601
+
+import java.time.{Duration, Period}
+
+import scala.util.{Failure, Success, Try}
+import cats.syntax.either._
+import cats.syntax.option._
+
+
+object IntervalDuration {
+
+  private val Regex = """(P.*?)(T.*)?""".r
+
+  val Zero = IntervalDuration(Period.ZERO, Duration.ZERO)
+
+
+  def apply(years: Int = 0, months: Int = 0, days: Int = 0,
+            hours: Int = 0, minutes: Int = 0, seconds: Int = 0): IntervalDuration = {
+    val dateDuration = Period.of(years, months, days)
+    val timeDuration = Duration.parse(s"PT${hours}H${minutes}M${seconds}S")
+    new IntervalDuration(dateDuration, timeDuration)
+  }
+
+  def apply(expression: String): Either[String, IntervalDuration] = {
+    expression match {
+      case Regex(dateDurationExpr, timeDurationExpr) =>
+        for {
+          dateDuration     <- parseDateDuration(expression, dateDurationExpr)
+          timeDuration     <- parseTimeDuration(expression, timeDurationExpr)
+          intervalDuration <- buildIntervalDuration(expression, dateDuration, timeDuration)
+        } yield intervalDuration
+
+      case _ => s"Wrong interval duration expression: $expression".asLeft
+    }
+  }
+
+  private def parseDateDuration(expression: String, durationExpr: String): Either[String, Option[Period]] = {
+    if (durationExpr.equalsIgnoreCase("P"))
+      none.asRight
+    else
+      Try(Period.parse(durationExpr)) match {
```

Comment:
you can use `Either.catchNonFatal(Period.parse(durationExpr))`

Reply:
Thanks for the suggestion, I didn't know it.